examples/simple_listener/simple_listener.c

```diff
@@ -170,90 +235,130 @@ void sigint_handler(int signum)
 #endif /* LIBSND */
 }
 
-int main(int argc, char *argv[])
+int init_mrp(struct mrp_listener_ctx *ctx, struct mrp_domain_attr *class_a, struct mrp_domain_attr *class_b)// struct mrp_domain_attr *class_a, struct mrp_domain_attr *class_a)
 {
-	char* file_name = NULL;
-	char* dev = NULL;
-	char errbuf[PCAP_ERRBUF_SIZE];
-	struct bpf_program comp_filter_exp;		/* The compiled filter expression */
-	char filter_exp[100];				/* The filter expression */
-	struct mrp_listener_ctx *ctx = malloc(sizeof(struct mrp_listener_ctx));
-	struct mrp_domain_attr *class_a = malloc(sizeof(struct mrp_domain_attr));
-	struct mrp_domain_attr *class_b = malloc(sizeof(struct mrp_domain_attr));
-	ctx_sig = ctx;
-	signal(SIGINT, sigint_handler);
-
-	int c,rc;
-	while((c = getopt(argc, argv, "hi:f:")) > 0)
-	{
-		switch (c)
-		{
-		case 'h':
-			help();
-			break;
-		case 'i':
-			dev = strdup(optarg);
-			break;
-		case 'f':
-			file_name = strdup(optarg);
-			break;
-		default:
-          		fprintf(stderr, "Unrecognized option!\n");
-		}
-	}
-
-	if ((NULL == dev) || (NULL == file_name))
-		help();
+	int rc;
 
 	rc = mrp_listener_client_init(ctx);
 	if (rc)
 	{
 		printf("failed to initialize global variables\n");
-		return EXIT_FAILURE;
+		return -1;
 	}
 
 	if (create_socket(ctx))
 	{
 		fprintf(stderr, "Socket creation failed.\n");
-		return errno;
+		return -1;
 	}
 
 	rc = mrp_monitor(ctx);
 	if (rc)
 	{
 		printf("failed creating MRP monitor thread\n");
-		return EXIT_FAILURE;
+		return -1;
 	}
-	rc=mrp_get_domain(ctx, class_a, class_b);
+	rc = mrp_get_domain(ctx, class_a, class_b);
 	if (rc)
 	{
 		printf("failed calling mrp_get_domain()\n");
-		return EXIT_FAILURE;
+		return -1;
 	}
 
 	printf("detected domain Class A PRIO=%d VID=%04x...\n",class_a->priority,class_a->vid);
 
 	rc = report_domain_status(class_a,ctx);
 	if (rc) {
 		printf("report_domain_status failed\n");
-		return EXIT_FAILURE;
+		return -1;
 	}
 
 	rc = join_vlan(class_a, ctx);
 	if (rc) {
 		printf("join_vlan failed\n");
-		return EXIT_FAILURE;
+		return -1;
+	}
+
+	return 0;
+}
+
+int main(int argc, char *argv[])
+{
+	char* file_name = NULL;
+	char* dev = NULL;
+	int sf_pcm_format = SF_FORMAT_PCM_24;
+	char errbuf[PCAP_ERRBUF_SIZE];
+	struct bpf_program comp_filter_exp;		/* The compiled filter expression */
+	char filter_exp[100];				/* The filter expression */
+	char dest_mac[30];
+	pcap_handler callback = pcap_callback;
+	struct mrp_listener_ctx *ctx = malloc(sizeof(struct mrp_listener_ctx));
+	struct mrp_domain_attr *class_a = malloc(sizeof(struct mrp_domain_attr));
+	struct mrp_domain_attr *class_b = malloc(sizeof(struct mrp_domain_attr));
+	ctx_sig = ctx;
+	signal(SIGINT, sigint_handler);
+
+	int c, rc;
+	while((c = getopt(argc, argv, "anhi:f:")) > 0)
+	{
+		switch (c)
+		{
+		case 'h':
+			help();
+			break;
+		case 'i':
+			dev = strdup(optarg);
+			break;
+		case 'f':
+			file_name = strdup(optarg);
+			break;
+		case 'n':
+			glob_no_srp = 1;
+			sprintf(dest_mac,"%s", static_dest_mac);
```

Comment:
I think it's fine, and the pull request has been open long enough. I'd be happy to merge another pull request that changes it to strcpy if someone wants to write it, though.